examples/file_body.hpp

```diff
@@ -36,8 +36,8 @@ struct file_body
 
     class writer
     {
-        std::size_t size_;
-        std::size_t offset_ = 0;
+        std::uint64_t size_;
+        std::uint64_t offset_ = 0;
         std::string const& path_;
         FILE* file_ = nullptr;
         char buf_[4096];
```

Comment:
I hadn't given it any thought. The implementation never makes copies or tries to move a writer, its an ephemeral object.